### PR TITLE
Add get-token flag to api-server start

### PR DESCRIPTION
**Pull request by rmn-boiko**
Related issue https://github.com/kamu-data/kamu-cli/issues/376

### CODE REVIEW

File: src/app/cli/src/explore/api_server.rs [hunk 1]
```diff
@@ -26,14 +30,16 @@ pub struct APIServer {
     task_executor: Arc<dyn TaskExecutor>,
     flow_service: Arc<dyn FlowService>,
     time_source: Arc<dyn SystemTimeSource>,
+    access_token: String,
 }
 
 impl APIServer {
-    pub fn new(
+    pub async fn new(
```
Comment (zaychenko-sergei) [inline src/app/cli/src/explore/api_server.rs:37]: 1) Can we avoid async new? It's not like a serious rule, but usually constructors should be sync and should not return errors. 

2) Perhaps, the token generation logic might be moved to command level. `APIServer` does not use it anyhow, it just generates it and provides a getter, but no other usage. This is an indication it's likely not owning this responsibility.

File: src/app/cli/src/explore/api_server.rs [hunk 2]
```diff
@@ -45,6 +51,23 @@ impl APIServer {
...
+
+        let auth_svc = base_catalog.get_one::<dyn AuthenticationService>().unwrap();
+        let access_token = auth_svc
+            .login(
+                &accounts::LOGIN_METHOD_PASSWORD.to_string(),
+                serde_json::to_string::<accounts::PasswordLoginCredentials>(&login_credentials)
+                    .unwrap(),
+            )
+            .await
+            .unwrap()
```
Comment (zaychenko-sergei) [inline src/app/cli/src/explore/api_server.rs:68]: Convert these unwraps into `int_err()` once you move the logic into command.
The tool should not simply die with a stack trace, instead we have a mechanism of internal errors that map to CLI error handlers.

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]

File: src/app/cli/src/explore/api_server.rs (hunk 1 above)
Comment (rmn-boiko) [reply src/app/cli/src/explore/api_server.rs]: Noticed for the future as well

File: src/app/cli/src/cli_commands.rs [hunk 3]
```diff
@@ -437,12 +437,17 @@ pub fn get_command(
...
+                    let current_account_subject = cli_catalog.get_one::<CurrentAccountSubject>()?;
+
                     Box::new(APIServerRunCommand::new(
                         base_catalog.clone(),
                         workspace_svc.is_multi_tenant_workspace(),
                         cli_catalog.get_one()?,
                         server_matches.get_one("address").map(|a| *a),
                         server_matches.get_one("http-port").map(|p| *p),
+                        server_matches.get_flag("get-token"),
+                        current_account_subject,
```
Comment (zaychenko-sergei) [inline src/app/cli/src/cli_commands.rs:450]: It's enough to write `cli_catalog.get_one()?`, the automatic type deduction will do the magic.

File: src/app/cli/src/commands/system_api_server_run_command.rs [hunk 4]
```diff
@@ -63,6 +75,28 @@ impl Command for APIServerRunCommand {
...
+        let auth_svc = self.catalog.get_one::<dyn AuthenticationService>().unwrap();
+        let access_token = auth_svc
+            .login(
+                &accounts::LOGIN_METHOD_PASSWORD.to_string(),
+                serde_json::to_string::<accounts::PasswordLoginCredentials>(&login_credentials)
+                    .int_err()?,
+            )
+            .await
+            .int_err()?
+            .access_token;
```
Comment (zaychenko-sergei) [inline src/app/cli/src/commands/system_api_server_run_command.rs:99]: The method `run` is getting a bit long, it basically doubled. Perhaps sub-methods for the large parts would make it more readable. At least a private method for the the new code.

**Review by zaychenko-sergei — APPROVED** [review]
Don't forget CHANGELOG.md

**Pull request merged**